pkg/build/builder/sti.go

```diff
@@ -53,7 +56,15 @@ func (s *STIBuilder) Build() error {
 	} else if s.build.Parameters.Source.Git.Ref != "" {
 		request.Ref = s.build.Parameters.Source.Git.Ref
 	}
-	glog.V(2).Infof("Creating a new STI builder with build request: %#v\n", request)
+	printRequest := *request
+	// If DockerCfgPath is provided in api.Request, then attempt to read the the
+	// dockercfg file and get the authentication for pulling the builder image.
+	if r, err := os.Open(request.DockerCfgPath); err == nil {
+		request.PullAuthentication = stidocker.GetImageRegistryAuth(r, request.BaseImage)
+		printRequest.PullAuthentication.Password = "[filtered]"
```

Comment:
@soltysh we can have nice printer for `Request` in STI which will solve this problem. Will add one and add todo here.